hazelcast/src/main/java/com/hazelcast/cache/impl/DeferredValue.java

```diff
@@ -84,20 +79,15 @@ public DeferredValue<V> shallowCopy() {
      * @return
      */
     public DeferredValue<V> shallowCopy(boolean resolved, SerializationService serializationService) {
-        if (this == NULL_VALUE) {
-            return NULL_VALUE;
-        }
         DeferredValue<V> copy = new DeferredValue<V>();
         if (serializedValueExists) {
             copy.serializedValueExists = true;
             copy.serializedValue = serializedValue;
+        } else if (!resolved && serializationService != null) {
+            copy.serializedValueExists = true;
+            copy.serializedValue = getSerializedValue(serializationService);
         }
-        if (!resolved && serializationService != null) {
-            if (!serializedValueExists) {
-                copy.serializedValueExists = true;
-                copy.serializedValue = getSerializedValue(serializationService);
-            }
-        } else if (valueExists) {
+        if (valueExists) {
```

Comment:
No need this else-if anymore --> `} else if (valueExists) {`?

Reply:
yes, reasoning is that if `this` deferred value already has both `serializedValue` & `value` populated, there is no reason to only copy `serializedValue` -- we can copy also the `value` and skip a deserialization.
Originally there was no `else` part here - it was added in [this change](https://github.com/hazelcast/hazelcast/commit/02b2b4556dbfd4551d4a8be403a7af2bf066b5a9#diff-45c896ad0074508c271e3b7d9546df843fed1deb227a8627cb987865251c2411R100)